### ISSUE TITLE
example MNIST does not work

### ISSUE THREAD

**Issue by LeoFanCE** [gcamfer/gym-classification#2]
Hello,
Thank you for your code. While I was running your MNIST example an error appeared.
![error](https://user-images.githubusercontent.com/94743844/150075430-b0330f28-add2-47bf-b302-1c6a02970b6a.PNG)

I added statements in your code to prevent the error since the truth value of an array with more than one element is ambiguous.
![add](https://user-images.githubusercontent.com/94743844/150076054-664e83a9-b535-477a-9a0e-3a81e2dfe526.PNG)

But as a result, the model could not converge like that you showed in your github page.
![result1](https://user-images.githubusercontent.com/94743844/150076363-2d83c9b1-e28d-4e4f-9d4c-11010eb3ad8b.PNG)

I wonder if you had the same problem or is there something that I've missed.

**Comment by gcamfer** [comment]
Hi, thank you very much for trying it and finding issues.
I don't really know why I returned a list of booleans, I guess my head was thinking about other stuff.  
I have fixed that problem.

Regarding the convergence, it seems to be a hyperparameter problem. Try reducing the exploration or reducing the learning rate. You can even rerun as random initialisation will get you to another point, but it's not very stable.

You can try the branch in this [Colab](https://colab.research.google.com/drive/1q_S48yxtzaOysSdgo1IYBdODn6CRkTeZ?usp=sharing) and next I'll merge with master.

Greetings

**Comment by JuiMahind** [comment]
Hello,
After execution of the code I am getting this error

UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use the `zero_division` parameter to control this behaviour.

this error is resulting in no f1 score for some labels.
can you give me any solution for this error
